graylog2-web-interface/src/theme/GraylogThemeProvider.tsx

```diff
@@ -15,51 +15,98 @@
  * <http://www.mongodb.com/licensing/server-side-public-license>.
  */
 import * as React from 'react';
-import { useCallback } from 'react';
+import { useEffect, useState } from 'react';
 import PropTypes from 'prop-types';
 import { ThemeProvider, DefaultTheme } from 'styled-components';
 
-import { breakpoints, colors, fonts, utils } from 'theme';
 import buttonStyles from 'components/graylog/styles/buttonStyles';
 import aceEditorStyles from 'components/graylog/styles/aceEditorStyles';
+import usePluginEntities from 'views/logic/usePluginEntities';
 
+import { breakpoints, colors, fonts, utils } from './index';
+import RegeneratableThemeContext from './RegeneratableThemeContext';
+import { Colors } from './colors';
+import { THEME_MODES, ThemeMode } from './constants';
 import useCurrentThemeMode from './UseCurrentThemeMode';
-import { THEME_MODES } from './constants';
+
+interface generateCustomFn {
```

Comment:
`generateCustomFn` sounds like it would be a function, but reading a bit further it looks like it's more like they are options.

Maybe renaming this will be a bit more clear?